### PR TITLE
Copy AES Key File on database copy

### PR DESCRIPTION
**Pull request by thewhitewizard**

### CODE REVIEW

File: src/main/java/com/iexec/sms/admin/AdminService.java [hunk 1]
```diff
@@ -206,38 +207,53 @@ boolean deleteBackupFileFromStorage(String storageLocation, String backupFileNam
...
+            final Path destinationDatabaseBackupFileLocation = Path.of(checkBackupFileLocation(
                     destinationStorageLocation + File.separator + destinationBackupFileName,
                     ERR_REPLICATE_OR_COPY_FILE_OUTSIDE_STORAGE));
 
+            // AES Key copy destination
+            final Path destinationAesKeyBackupFileLocation = Path.of(destinationDatabaseBackupFileLocation.toString() + AES_KEY_FILENAME_EXTENSION);
+
             // Check that we are not trying to overwrite a file
-            if (destinationBackupFileLocation.toFile().exists()) {
+            if (destinationDatabaseBackupFileLocation.toFile().exists() || destinationAesKeyBackupFileLocation.toFile().exists()) {
```
Comment (mcornaton) [inline src/main/java/com/iexec/sms/admin/AdminService.java:236]: The error should be more specific about which file already exists.

**Review by jbern0rd — APPROVED** [review]

File: src/main/java/com/iexec/sms/admin/AdminService.java [hunk 2]
```diff
@@ -206,44 +202,88 @@ boolean deleteBackupFileFromStorage(String storageLocation, String backupFileNam
...
+    }
+
+    /**
+     * Checks if the destination does not exist and throws an exception if it does
+     *
+     * @param destination         The destination to check
+     * @param adminOperationError The custom error message
+     * @throws IOException If the destination already exist
+     */
+    private void checkDestinationFileNotExist(Path destination, AdminOperationError adminOperationError) throws IOException {
```
[the remark this answers is not in the document]
Comment (jbern0rd) [reply src/main/java/com/iexec/sms/admin/AdminService.java:258]: I have the impression that there are 2 points of view here.
If we think the Does of DoesNotExist, the first variant becomes acceptable.
I don't really have a preference here. But if there is a rule, we may want to write it somewhere.

[the remark this answers is not in the document]
Comment (mcornaton) [reply src/main/java/com/iexec/sms/admin/AdminService.java]: You're right. I may have typed too fast, trying to standardize with my other comment. Current version is fine :)

[the remark this answers is not in the document]
Comment (thewhitewizard) [reply src/main/java/com/iexec/sms/admin/AdminService.java]: so i need to rollback ?

**Review by mcornaton — APPROVED** [review]
Thanks!

**Pull request merged**